styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java

```diff
@@ -44,6 +50,35 @@ private KubernetesPodEventTranslator() {
   private static final Predicate<ContainerStatus> IS_STYX_CONTAINER =
       (cs) -> KubernetesDockerRunner.STYX_RUN.equals(cs.getName());
 
+  @AutoMatter
+  private interface TerminationLogMessage {
+    String componentId();
+    String workflowId();
+    String parameter();
+    String executionId();
+    String event();
+    int exitCode();
+  }
+
+  private static int getExitCode(Pod pod, ContainerStatus status) {
+    if ("true".equals(pod.getMetadata().getAnnotations().get(DOCKER_TERMINATION_LOGGING_ANNOTATION))) {
+      try {
+        final TerminationLogMessage message = Json.OBJECT_MAPPER.readValue(
+            status.getState().getTerminated().getMessage(),
```

Comment:
This is what it might look like:

```
"containerStatuses": [
    {
      "containerID": "docker:\/\/492ebc331bb8397d8bf499898b116967e1038ef140db053e7b48979469655407",
      "image": "registry.spotify.net\/spotify\/gcp-migrate:fbdd6a8-1486977920410",
      "imageID": "docker:\/\/sha256:b742f831640398df796dc8cd15c30e82dcca1642461cb267f8c0c5cbd26ef86e",
      "lastState": {
        
      },
      "name": "styx-run",
      "ready": false,
      "restartCount": 0,
      "state": {
        "terminated": {
          "containerID": "docker:\/\/492ebc331bb8397d8bf499898b116967e1038ef140db053e7b48979469655407",
          "exitCode": 20,
          "finishedAt": "2017-03-09T14:14:25Z",
          "reason": "Error",
          "startedAt": "2017-03-09T14:13:06Z"
        }
      }
    }
  ],
```